p3/plot/_navchart.py

```diff
@@ -53,6 +46,15 @@ def navchart(pp, cd, eff=None, size=(5, 5), goal=None, **kwargs):
               - Type
               - Description
 
+            * - `size`
+              - 2-tuple of floats, default: (5, 5)
+              - The size of the plot, in backend-specific units.
+
+            * - `goal`
+              - tuple
+              - User-defined goal, expressed as (convergence, portability).
+                The region between this point and (1, 1) will be highlighted.
```

Comment:
Size and goal are not yet supported by the PGFPlots backend. That said, the great power of the PGFPlots backend is that it can be customised infinitely after generation by editing the TeX directly. 

So, the point here was just such that the things required by all backends are given to the `cascade` and `navchart` functions, and any other arguments that are backend-specific are given in `kwargs`.

Reply:
I think the way things are documented now, a user would expect all backends to support size and goal.  The other two things in this table (i.e., legend and style) are supported by both backends, and there's no indication that size and goal are matplotlib-specific.

I'm not sure about size yet, but I think the pgfplots backend should (eventually) support goal -- it's a useful thing to include on these sorts of plots.

I understand that supporting both of these options would be a big ask for this PR, so instead... Could you please add code to raise a `NotImplementedError` if the pgfplots backend contains "goal" or "size"?  That way, users will get a meaningful error message until we implement things.